s2nTests/docker/s2n.dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:22.04
+FROM ubuntu:24.04
 
 RUN apt-get update -y -q && \
     apt-get install -y software-properties-common && \
```

Comment:
GitHub won't let me highlight it, but the use of `libncurses5` below should be replaced with `libncurses6`.